ingestion/src/metadata/ingestion/source/database/postgres/utils.py

```diff
@@ -347,3 +355,34 @@ def get_view_definition(
         schema=schema,
         query=POSTGRES_VIEW_DEFINITIONS,
     )
+
+
+def get_postgres_version(engine) -> Optional[str]:
+    """
+    return the postgres version in major.minor.patch format
+    """
+    try:
+        results = engine.execute(POSTGRES_GET_SERVER_VERSION)
+        for res in results:
+            version_string = str(res[0])
+            opening_parenthesis_index = version_string.find("(")
+            if opening_parenthesis_index != -1:
+                return version_string[:opening_parenthesis_index].strip()
+            return version_string
+    except Exception as err:
+        logger.warning(f"Unable to fetch the Postgres Version - {err}")
+        logger.debug(traceback.format_exc())
+    return None
+
+
+def get_postgres_time_column_name(engine) -> str:
+    """
+    Return the correct column name for the time column based on postgres version
+    """
+    time_column_name = "total_exec_time"
+    postgres_version = get_postgres_version(engine)
+    if postgres_version and version.parse(postgres_version) < version.parse(
+        INCOMPATIBLE_POSTGRES_VERSION
```

Comment:
nit - is INCOMPATIBLE the right name here? that version is still compatible for us right?

Reply:
yea, let me rename to a meaningful name